SparkPost/2web2ui
# AC-564: Invite subaccount reporting users

# What I Did
 - Enabled optional subaccount assignment for reporting users
 - Updated the role description copy on the user invite page

# How To Test
 - Use an account with subaccounts
 - Flip the `subaccount_reporting` feature flag (see docs/feature-flags.md)
 - Manage users -> Invite User
 - Note the new subaccount checkbox and typeahead when selecting the new role
 - Checkout the user edit page too ;)

# Todo
 - [x] Prototype review feedback
 - [x] Complete implementation
 - [x] Check the user edit page for pre-release corner cases, since it shares `RoleRadioGroup`
